tests/Engine/Unit/Segments/SegmentModelsTest.php

```diff
@@ -74,6 +74,7 @@ public function conditionParametersTraitValues()
             [SegmentConditions::REGEX, 'foo', '[a-z]+', true],
             [SegmentConditions::REGEX, 'FOO', '[a-z]+', false],
             [SegmentConditions::REGEX, '1.2.3', '\\d', true],
+            [SegmentConditions::REGEX, 123, '^\\d*[13579]$', true],
```

Comment:
I added a test that check that an Integer trait value can be compared against a regex